tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go

```diff
@@ -273,9 +273,70 @@ func (a *awsProvider) findOrCreateSg(infraID string, vpc *ec2.Vpc) (string, erro
 	if err != nil {
 		return a.createWindowsWorkerSg(infraID, vpc)
 	}
+
+	// Additional check for existing security group
+	err = a.verifyOrCreateWinrmPort(sgID)
+	if err != nil {
+		return "", nil
+	}
+
 	return sgID, nil
 }
 
+// Find the winrm port 5986 for the security group. If the port doesn't not exist in the security group
+// notify the user and generate one for the user.
+func (a *awsProvider) verifyOrCreateWinrmPort(sgId string) error {
```

Comment:
Let's not add 2 different things into a single function, decompose this into 2 fns
1st function should check for the existence of SG and inbound rule and should return true or false
2nd function should just create it.

Reply:
Thanks for your comment.

I have split the function into two functions. One is `checkSgWinrmPort` which checks the winrm port and returns a boolean, the other one `createWinrmPortToSg` which adds the winrm port to the Security group. The boolean check result is taken care of in the `findOrCreateSg` function.

Manual local test passed. Please reviewed it. @ravisantoshgudimetla